src/mgr/PyModules.cc

```diff
@@ -664,3 +666,33 @@ PyObject *PyModules::get_context()
   return capsule;
 }
 
+static void _list_modules(
+  const std::string path,
+  std::set<std::string> *modules)
+{
+  DIR *dir = opendir(path.c_str());
+  if (!dir) {
+    return;
+  }
+  struct dirent *entry = NULL;
+  while ((entry = readdir(dir)) != NULL) {
+    string n(entry->d_name);
+    string fn = path + "/" + n;
+    struct stat st;
+    int r = ::stat(fn.c_str(), &st);
+    if (r == 0 && S_ISDIR(st.st_mode)) {
+      string initfn = fn + "/__init__.py";
```

Comment:
Should check for "module.py" -- we might have an __init__.py in a directory that was just e.g. shared code in a module.

The fixed module.py name is a bit hacky but probably actually helps in this case.